File: assignments/assignment-1-solution/src-csharp/CrazyEights/Cards/UnselectedCard.cs
using CrazyEights.Domain;

namespace CrazyEights.Cards;

/// <summary>
/// Null-object card used when no card is selected.
/// </summary>
public class UnselectedCard : ICard
{
    public static readonly UnselectedCard Instance = new();

    public RankType Rank => RankType.NotSet;

    public SuitType Suit => SuitType.NotSet;

    public int CardId => -1;

    public string GetDescription()
    {
        return "Unselected Card";
    }

    public bool IsSelectable => false;
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/cards/UnselectedCard.java
package edu.kennesaw.crazy8s.cards;

import edu.kennesaw.crazy8s.domain.RankType;
import edu.kennesaw.crazy8s.domain.SuitType;

/**
 * Null-object card used when no card is selected.
 */
public final class UnselectedCard implements Card {
    private static final UnselectedCard INSTANCE = new UnselectedCard();

    private UnselectedCard() {
    }

    public static UnselectedCard getInstance() {
        return INSTANCE;
    }

    @Override
    public RankType getRank() {
        return RankType.NOT_SET;
    }

    @Override
    public SuitType getSuit() {
        return SuitType.NOT_SET;
    }

    @Override
    public int getCardId() {
        return -1;
    }

    @Override
    public String getDescription() {
        return "Unselected Card";
    }

    @Override
    public boolean isSelectable() {
        return false;
    }
}
